### PR TITLE
KAFKA-6752: Enable unclean leader election metric

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@junrao pls take a look when you get a chance

File: core/src/main/scala/kafka/controller/PartitionStateMachine.scala [hunk 1]
```diff
@@ -435,9 +435,10 @@ class PartitionStateMachine(config: KafkaConfig,
 }
 
 object PartitionLeaderElectionAlgorithms {
-  def offlinePartitionLeaderElection(assignment: Seq[Int], isr: Seq[Int], liveReplicas: Set[Int], uncleanLeaderElectionEnabled: Boolean): Option[Int] = {
+  def offlinePartitionLeaderElection(assignment: Seq[Int], isr: Seq[Int], liveReplicas: Set[Int], uncleanLeaderElectionEnabled: Boolean, controllerContext: ControllerContext): Option[Int] = {
     assignment.find(id => liveReplicas.contains(id) && isr.contains(id)).orElse {
       if (uncleanLeaderElectionEnabled) {
+        controllerContext.stats.uncleanLeaderElectionRate.mark()
```
Comment (junrao) [inline core/src/main/scala/kafka/controller/PartitionStateMachine.scala]: We probably only want to mark the metric if assignment.find(liveReplicas.contains) is not empty.

**Review by junrao — COMMENTED** [review]
@omkreddy : Thanks for the patch. LGTM. Just a minor comment below.

File: core/src/main/scala/kafka/controller/PartitionStateMachine.scala (hunk 1 above)
Comment (omkreddy) [reply core/src/main/scala/kafka/controller/PartitionStateMachine.scala]: @junrao Thanks for the review. Updated the PR.

**Review by junrao — APPROVED** [review]
@omkreddy : Thanks for the updated patch. LGTM

**Comment by junrao** [comment]
merged to trunk and 1.1.

**Pull request merged**